pkg/plugins/resources/k8s/store.go

```diff
@@ -160,9 +160,34 @@ func (s *KubernetesStore) List(ctx context.Context, rs core_model.ResourceList,
 	if err := s.Converter.ToCoreList(obj, rs, predicate); err != nil {
 		return errors.Wrap(err, "failed to convert k8s model into core counterpart")
 	}
+
+	total, err := s.countK8sResources(ctx, rs, opts.Mesh)
+	if err != nil {
+		return err
+	}
+	rs.GetPagination().SetTotal(uint32(total))
+
 	return nil
 }
 
+func (s *KubernetesStore) countK8sResources(ctx context.Context, rs core_model.ResourceList, mesh string) (int, error) {
+	obj, err := s.Converter.ToKubernetesList(rs)
+	if err != nil {
+		return 0, errors.Wrapf(err, "failed to convert core list model of type %s into k8s counterpart", rs.GetItemType())
+	}
+
+	if err := s.Client.List(ctx, obj, &kube_client.ListOptions{}); err != nil {
+		return 0, errors.Wrap(err, "failed to list k8s resources")
+	}
+
+	total := 0
+	if total, err = s.Converter.CountMeshItems(obj, rs.NewItem(), mesh); err != nil {
+		return 0, errors.Wrap(err, "failed to convert k8s model into core counterpart")
+	}
+
+	return total, nil
```

Comment:
How about simplifying it a bit. I don't feel like `CountMeshItems` should be in `Converter`. We can do this without converting items
```
func (s *KubernetesStore) countK8sResources(ctx context.Context, rs core_model.ResourceList, mesh string) (int, error) {
	obj, err := s.Converter.ToKubernetesList(rs)
	if err != nil {
		return 0, errors.Wrapf(err, "failed to convert core list model of type %s into k8s counterpart", rs.GetItemType())
	}

	if err := s.Client.List(ctx, obj, &kube_client.ListOptions{}); err != nil {
		return 0, errors.Wrap(err, "failed to list k8s resources")
	}

	total := 0
	for _, item := range obj.GetItems() {
		if item.GetMesh() == mesh {
			total++
		}
	}

	return total, nil
}
```

Reply:
Yep, my proposed approach was ugly. I applied your way with a small optimization when the mesh name is empty. Thanks!